src/test/kotlin/org/rust/ide/inspections/RsWrongTypeArgumentsNumberInspectionTest.kt

```diff
@@ -188,4 +188,72 @@ class RsWrongTypeArgumentsNumberInspectionTest : RsInspectionsTestBase(RsWrongTy
             err1: Foo<u32, i32>,
         }
     """)
+
+    fun `test add arguments missing arguments`() = checkFixByText("Add missing type arguments", """
+        struct S<T> { t: T }
+
+        fn main() {
+            let x: <error descr="Wrong number of type arguments: expected 1, found 0 [E0107]">S/*caret*/</error>;
+        }
+    """, """
+        struct S<T> { t: T }
+
+        fn main() {
+            let x: S<T>;
+        }
+    """)
+
+    fun `test add arguments copy existing arguments`() = checkFixByText("Add missing type arguments", """
+        struct S<T, R> { t: T, r: R }
+
+        fn main() {
+            let x: <error descr="Wrong number of type arguments: expected 2, found 1 [E0107]">S<u32/*caret*/></error>;
+        }
+    """, """
+        struct S<T, R> { t: T, r: R }
+
+        fn main() {
+            let x: S<u32, R>;
+        }
+    """)
+
+    fun `test add arguments keep lifetimes`() = checkFixByText("Add missing type arguments", """
+        struct S<'a, T, R> { t: &'a T, r: R }
+
+        fn foo<'a>(x: &'a u32) {
+            let x: <error descr="Wrong number of type arguments: expected 2, found 1 [E0107]">S<'a, u32/*caret*/></error>;
+        }
+    """, """
+        struct S<'a, T, R> { t: &'a T, r: R }
+
+        fn foo<'a>(x: &'a u32) {
+            let x: S<'a, u32, R>;
+        }
+    """)
+
+    fun `test add arguments keep associated types`() = checkFixByText("Add missing type arguments", """
+        trait S<A, B> {
+            type Item;
+        }
+
+        fn main() {
+            let x: <error descr="Wrong number of type arguments: expected 2, found 1 [E0107]">S<u32, Item=u32/*caret*/></error>;
+        }
+    """, """
+        trait S<A, B> {
+            type Item;
+        }
+
+        fn main() {
+            let x: S<u32, B, Item=u32>;
+        }
+    """)
+
+    fun `test add arguments ignore type parameters with a default`() = checkFixIsUnavailable("Add missing type arguments", """
+        struct S<A, B=u32, C=u32>(A, B, C);
+
+        fn main() {
+            let x: S<u32, u32>;
```

Comment:
Hm, all tests of new quick fix use only struct although it should support function and method calls. Let's add necessary tests to be sure that everything works as expected